### PR TITLE
perf(box): improve performance for boxes

### PR DESCRIPTION
**Pull request by quanho**
This PR is to:
1. only add markers (line, median, etc.) used,
2. optimize the loop of adding markers.

### CODE REVIEW

File: packages/picasso.js/src/core/chart-components/box/box-shapes.js [hunk 1]
```diff
@@ -208,114 +209,174 @@ export function buildShapes({
...
+  let boxWidth;
+  let boxPadding;
+  let boxCenter;
+  let isLowerOutOfBounds;
+  let isHigherOutOfBounds;
+  let isOutOfBounds;
+  const numKeys = keys ? keys.length : 0;
+  const numNonOobKeys = nonOobKeys ? nonOobKeys.length : 0;
+
+  function addBox() {
```
Comment (cbt1) [inline packages/picasso.js/src/core/chart-components/box/box-shapes.js]: Why not go all the way and break out these "add" functions from buildShapes? As buildShapes is now a ~200 lines function with a lot of complex logic that should be tested separately.

File: packages/picasso.js/src/core/chart-components/box/box-shapes.js [hunk 2]
```diff
@@ -208,114 +209,174 @@ export function buildShapes({
...
+        item: minorItem, value: 0, boxCenter, rendWidth, rendHeight, flipXY, symbol
+      }));
+    } else if (isHigherOutOfBounds) {
+      children.push(oob({
+        item: minorItem, value: 1, boxCenter, rendWidth, rendHeight, flipXY, symbol
+      }));
+    }
+  }
+
+  const addMarkerList = {
```
Comment (cbt1) [inline packages/picasso.js/src/core/chart-components/box/box-shapes.js]: Declare closer to usage.

**Comment by cbt1** [comment]
Not a single new test?

**Comment by quanho** [comment]
> Not a single new test?

Added unit tests.

**Pull request merged**